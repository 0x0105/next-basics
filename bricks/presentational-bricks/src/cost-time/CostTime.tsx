import React from "react";
import { costTime } from "@next-libs/datetime";

interface CostTimeProps {
  startTime?: string | number;
  endTime?: string | number;
  cost?: number;
}

export function CostTime({
  cost,
  startTime,
  endTime,
}: CostTimeProps): React.ReactElement {
  return <span>{costTime(cost, startTime, endTime)}</span>;
}
