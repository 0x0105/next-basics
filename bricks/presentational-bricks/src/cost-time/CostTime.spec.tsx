import React from "react";
import { shallow } from "enzyme";
import { CostTime } from "./CostTime";

describe("CostTime", () => {
  it("should work", () => {
    const wrapper = shallow(<CostTime cost={123} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should work when cost is undefined", () => {
    const wrapper = shallow(
      <CostTime startTime="2019-10-19 18:13:18" endTime="2019-10-19 18:13:21" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
