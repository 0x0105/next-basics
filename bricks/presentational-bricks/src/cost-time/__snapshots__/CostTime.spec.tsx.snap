// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CostTime should work 1`] = `
<span>
  0.1 秒
</span>
`;

exports[`CostTime should work when cost is undefined 1`] = `
<span>
  3 秒
</span>
`;
